### DISCUSSION TITLE
Unable to load custom yolov5 tflite model in the camera

### DISCUSSION THREAD

**Discussion by vishalkumarbit** [AxisCommunications/acap-computer-vision-sdk-examples#139] [General] [answered]
I am trying to load a custom model in the camera we are facing some issues the error is: 
<_InactiveRpcError of RPC that terminated with: 
status = StatusCode.UNAVAILABLE, 
details = "failed to connect to all addresses"

We believe this is because the inference container is not able to load the model. The common known issues page(https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/111) of the SDK says that if we face this error, we should leave the app running for 3-5 minutes as it takes time to load a model for the first time. But we were seeing this error even after 10 mins, we also followed the next steps. i.e. When we try "docker -H ps" the apac-runtime container is not running and docker compose logs are empty.

The journalctl -u larod command in the camera SSH console gives the following output
Mar 23 13:22:17 axis-b8a44f6cba26 sh[243722]: ERROR: Failed to run graph
Mar 23 13:22:17 axis-b8a44f6cba26 larod[243722]: Node number 429 (Vx Delegate) failed to invoke.
Mar 23 13:22:18 axis-b8a44f6cba26 larod[243722]: Could not send load model result: Session has expired
Mar 23 13:22:18 axis-b8a44f6cba26 larod[243722]: Could not send delete model result: Session has expired
Mar 23 13:22:33 axis-b8a44f6cba26 larod[243722]: Created a new session ID: 3, client: :1.1827

<img width="1426" alt="Screen Shot 2023-03-23 at 4 46 06 PM" src="https://user-images.githubusercontent.com/59842524/227432513-936874cc-ecce-4c17-82ce-d884e1865228.png">

**Comment by Corallo** [comment]
Hello!
Yes, it seems that your model is not compatible, after running the example, ssh into the camera a run `journalctl -u larod`.
This might give you a bit more information about the issue with your model.

Or, by loading the model on [netron.app](https://netron.app/) you can search for 429 and see what is the layer that cause the problem. 

In general, our devices support only tensorflow lite operators, try to stick with them to avoid issues:
https://www.tensorflow.org/lite/guide/ops_compatibility
https://www.tensorflow.org/lite/guide/ops_select

Edit: 
re-reading your question, I noticed missed a couple of points.

- Regarding what mentioned in [#111](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/discussions/111), you are in the 1st scenario, where the model actually crash, in this case waiting doesn't help. Sometimes, if you don't see any error log, you just have to wait for the model to be internally converted, but this is not your case.

- I noticed now that in the title you mentioned that your model is yolov5. Is it the ultralytics one?
Is it the object detector or the segmentation model?
The segmentation model is not currently supported, we are currently investigating, but it will take a while.
If you are trying the object detection model, that should at least be loaded correctly, did you export it using their export script?
`python export.py --weights yolov5s.pt --int8 --include tflite`

**Comment by param-esec** [comment]
Hi,
I trained my model with [yolov5](https://raw.githubusercontent.com/param-esec/tfmodel/main/yolov5.tflite) and then converted it to tflite using `python export.py --weights yolov5s.pt --int8 --include tflite`. 
However i got the following error when i tested it on my artpec8 camera. There seems to be quantization problem, could you please tell me how to solve the issue
![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/95c0f1c0-f073-40ae-a6b7-ea05bfd96e1c)

`journalctl -u larod` report
![image](https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/assets/137760057/e6efe577-b8f6-4285-8fed-312f7efaa76f)


Thank you in advance.

**Reply by Corallo** [reply]
Hi @Corallo 

We would suggest not to use yolo, as the quantization that they do is not suitable for A8, causing these kinds of problems. https://github.com/AxisCommunications/acap-computer-vision-sdk-examples/issues/140

If you want to investigate, you could try to load the model on netron.app and see what is that node 408 that the larod process is complaining about in this case

**Comment by Corallo** [comment, marked as answer]
Hi again, 
It is now possible to run Yolov5 on Artpec-8 on devices running Axis OS >= 11.7
See the guide [here](https://github.com/AxisCommunications/axis-model-zoo/blob/main/docs/yolov5-on-artpec8.md)
